.github/build_and_test.yml

```diff
@@ -0,0 +1,33 @@
+name: Build and Test
+
+on: [pull_request, workflow_dispatch, workflow_call]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: develop
+    permissions:
+      actions: write
+      id-token: write
+      contents: read
+      checks: write
+      packages: write
+      pull-requests: write
+    strategy:
+      matrix:
+        step: ['format:check', 'lint:check', 'build', 'test:unit']
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: .nvmrc
+          cache: 'yarn'
+      - run: yarn install
+      - run: yarn ${{ matrix.step }}
+        env:
+          ARNS_CONTRACT_TX_ID: ${{ vars.ARNS_CONTRACT_TX_ID }}
+      - name: Upload to Codecov
+        if: matrix.step == 'test:unit'
+        uses: codecov/codecov-action@v3
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
token has been added

Reply:
Should I re-add this with the pill on the read me?

Reply:
Don't worry about the pill